src/Data/ProgrammesDb/Entity/Ancestry.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\Entity;
+
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\Entity\EntityInterfaces\MemberOfGroupInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\Entity\EntityInterfaces\PromotableInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\Entity\EntityInterfaces\RelatedLinkContextInterface;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Timestampable\Traits\TimestampableEntity;
+
+/**
+ * @ORM\Entity(repositoryClass="BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\ImageRepository")
+ */
+class Ancestry implements MemberOfGroupInterface, RelatedLinkContextInterface, PromotableInterface
+{
+    use TimestampableEntity;
+    use Traits\IsEmbargoedTrait;
+    use Traits\PartnerPidTrait;
+    use Traits\SynopsesTrait;
+
+    /**
+     * @var int|null
+     *
+     * @ORM\Id()
+     * @ORM\Column(type="integer", nullable=true, unique=true)
+     * @ORM\ManyToOne(targetEntity="CoreEntity")
+     * @ORM\JoinColumn(nullable=true)
+     */
```

Comment:
We will need to make all integer values non-nullable for the ancestry table:
```suggestion
    /**
     * @var int     *
     * @ORM\Id()
     * @ORM\Column(type="integer", nullable=false, unique=true)
     * @ORM\ManyToOne(targetEntity="CoreEntity")
     * @ORM\JoinColumn(nullable=false)
     */
```